contracts/factories/BridgeFactory.sol

```diff
@@ -0,0 +1,274 @@
+    periodMintLimit = _newPeriodMintLimit;
+  }
+
+  /**
+   * @notice Send current ARC balance of conquest contract to recipient
+   * @param _recipient Address where the currency will be sent to
+   * @param _data      Data to pass with transfer function
+   */
+  function withdraw(address _recipient, bytes calldata _data) external onlyOwnerTier(HIGHEST_OWNER_TIER) {
+    require(_recipient != address(0x0), "BridgeFactory#withdraw: INVALID_RECIPIENT");
+    uint256 thisBalance = arcadeumCoin.balanceOf(address(this), arcadeumCoinID);
+    arcadeumCoin.safeTransferFrom(address(this), _recipient, arcadeumCoinID, thisBalance, _data);
+  }
+
+
+  /***********************************|
+  |      Receiver Method Handler      |
+  |__________________________________*/
+
+  /**
+   * @notice Prevents receiving Ether or calls to unsuported methods
+   */
+  fallback () external {
+    revert("BridgeFactory#_: UNSUPPORTED_METHOD");
+  }
+
+  /**
+   * @notice Burns SW asset when received, or store ARC
+   * @dev Make sure to send the right assets and the correct amount 
+   *      as it's not validated in this contract.
+   * @param _id      Id of Token being transferred
+   * @param _amount  Amount of Token _id being transferred
+   */
+  function onERC1155Received(
+    address, // _operator
+    address, // _from 
+    uint256 _id, 
+    uint256 _amount, 
+    bytes memory // _data
+  )
+    public override returns(bytes4)
+  {
+    if (msg.sender == address(arcadeumCoin)) {
+      // Do nothing, just store the ARC until withdrawl
+      require(_id == arcadeumCoinID, "BridgeFactory#onERC1155Received: INVALID_ARC_ID");
+
+    } else if (msg.sender == address(skyweaverAssets)) {
+      // Burn asset received
+      ISkyweaverAssets(msg.sender).burn(_id, _amount);
+
+    } else {
+      revert("BridgeFactory#onERC1155Received: INVALID_TOKEN");
+    }
+
+    return IERC1155TokenReceiver.onERC1155Received.selector;
+  }
+
+  /**
+   * @notice Burns SW assets when received, or store ARC
+   * @dev Make sure to send the right assets and the correct amount 
+   *      as it's not validated in this contract.
+   * @param _ids     An array containing ids of each Token being transferred
+   * @param _amounts An array containing amounts of each Token being transferred
+   */
+  function onERC1155BatchReceived(
+    address, // _operator
+    address, // _from
+    uint256[] memory _ids,
+    uint256[] memory _amounts,
+    bytes memory // _data
+  )
+    public override returns(bytes4)
+  {
+    if (msg.sender == address(skyweaverAssets)) {
+      // Burn assets received
+      ISkyweaverAssets(msg.sender).batchBurn(_ids, _amounts);
+
+    // Arc tribute
+    } else if (msg.sender == address(arcadeumCoin)) {
+      // Do nothing, just store the ARC until withdrawl
+      require(_ids[0] == arcadeumCoinID, "BridgeFactory#onERC1155BatchReceived: INVALID_ARC_ID");
+      require(_ids.length == 1, "BridgeFactory#onERC1155BatchReceived: INVALID_ARRAY_LENGTH");
```

Comment:
What do you think of removing these two checks? Whatever tokens come from `arcadeumCoin` contract can be considered valid and in the withdraw method we can add an ID parameter, in case someone sends the wrong token (if we have other ids in the future). Would save an SLOAD on ARC deposits.

Reply:
Alternatively we can make all of this immutable:

```
  // Token information
  ISkyweaverAssets internal skyweaverAssets; // ERC-1155 Skyweaver assets contract
  IERC1155 internal arcadeumCoin;            // ERC-1155 Arcadeum Coin contract
  uint256 internal arcadeumCoinID;           // ID of ARC token in respective ERC-1155 contract
```

Reply:
Oh neat, that's a great idea.